moveit_ros/planning/planning_pipeline/src/planning_pipeline_middleware_handle.cpp

```diff
@@ -0,0 +1,261 @@
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Anthony Baker */
+
+#include <moveit/planning_pipeline/planning_pipeline_middleware_handle.hpp>
+#include <moveit/planning_pipeline/planning_pipeline.h>
+#include <rclcpp/rclcpp.hpp>
+#include <boost/algorithm/string/join.hpp>
+#include <string>
+
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit.ros_planning.planning_pipeline_middleware_handle");
+
+namespace planning_pipeline
+{
+namespace
+{
+static const auto SUBSCRIPTION_QOS = rclcpp::QoS(10);
+}
+
+PlanningPipelineMiddlewareHandle::PlanningPipelineMiddlewareHandle(const rclcpp::Node::SharedPtr& node) : node_(node)
+{
+}
+
+void PlanningPipelineMiddlewareHandle::createDisplayPathPublisher(const std::string& topic, bool flag)
+{
+  display_path_publisher_ = node_->create_publisher<moveit_msgs::msg::DisplayTrajectory>(topic, SUBSCRIPTION_QOS);
+}
+
+void PlanningPipelineMiddlewareHandle::createReceivedRequestPublisher(const std::string& topic, bool flag)
+{
+  received_request_publisher_ = node_->create_publisher<moveit_msgs::msg::MotionPlanRequest>(topic, SUBSCRIPTION_QOS);
+}
+
+void PlanningPipelineMiddlewareHandle::createContactsPublisher(const std::string& topic, bool flag)
+{
+  contacts_publisher_ = node_->create_publisher<visualization_msgs::msg::MarkerArray>(topic, SUBSCRIPTION_QOS);
+}
+
+std::string PlanningPipelineMiddlewareHandle::getContactsTopicName() const
+{
+  return contacts_publisher_->get_topic_name();
+}
+
+void PlanningPipelineMiddlewareHandle::resetDisplayPathPublisher()
+{
+  display_path_publisher_.reset();
+}
+
+void PlanningPipelineMiddlewareHandle::resetReceivedRequestPublisher()
+{
+  received_request_publisher_.reset();
+}
+
+void PlanningPipelineMiddlewareHandle::resetContactsPublisher()
+{
+  contacts_publisher_.reset();
+}
+
+bool PlanningPipelineMiddlewareHandle::has_parameter(const std::string& name) const
+{
+  return node_->has_parameter(name);
+}
+
+void PlanningPipelineMiddlewareHandle::get_parameter(const std::string& pluginName, std::string& plugin) const
+{
+  node_->get_parameter(pluginName, plugin);
+}
+
+void PlanningPipelineMiddlewareHandle::createPlannerPlugin(const std::string& name,
+                                                           const moveit::core::RobotModelConstPtr& robot_model)
+{
+  try
+  {
+    planner_plugin_loader_.reset(new pluginlib::ClassLoader<planning_interface::PlannerManager>(
```

Comment:
Should be replaceable with `std::make_unique<pluginlib::ClassLoader<planning_interface::PlannerManager>(...)`